docs/aws/audit/redshiftmonitoring/rules/redshift_cluster_encrypted_with_kms_customer_master_keys.mdx

```diff
@@ -23,6 +23,208 @@ NIST, SOC2, NISTCSF, PCIDSS
+2. Navigate to the Redshift dashboard by selecting "Services" from the top menu, then selecting "Redshift" under the "Database" category.
+3. In the Redshift dashboard, select "Clusters" from the left-hand menu. This will display a list of all your Redshift clusters.
+4. For each cluster, check the "KMS Key ID" column. If this column is empty or the key ID listed is not a KMS CMK, then the Redshift cluster is not encrypted with a KMS CMK.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the Redshift clusters in your AWS account. You can do this by using the AWS CLI command `aws redshift describe-clusters`. This command will return a JSON output with all the Redshift clusters.
+
+    ```
+    aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier]' --output text
+    ```
+
+2. Once you have the list of Redshift clusters, you can check the encryption settings of each cluster. You can do this by using the AWS CLI command `aws redshift describe-clusters --cluster-identifier <cluster-name>`. Replace `<cluster-name>` with the name of your Redshift cluster. This command will return a JSON output with the details of the specified Redshift cluster.
+
+    ```
+    aws redshift describe-clusters --cluster-identifier <cluster-name> --query 'Clusters[*].[KmsKeyId]' --output text
+    ```
+
+3. If the `KmsKeyId` field is empty or null, it means that the Redshift cluster is not encrypted with a KMS CMK. If the `KmsKeyId` field contains a value, it means that the Redshift cluster is encrypted with a KMS CMK.
+
+4. Repeat step 2 and 3 for each Redshift cluster in your AWS account. This way, you can check the encryption settings of all Redshift clusters in your AWS account.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services including Redshift.
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+3. Create a Redshift client using the session.
+
+```python
+redshift = session.client('redshift')
+```
+
+4. Use the `describe_clusters` method to retrieve information about all Redshift clusters. Then, check the `KmsKeyId` attribute for each cluster. If this attribute is not present or is not a valid KMS CMK, the cluster is not encrypted with a KMS CMK.
+
+```python
+clusters = redshift.describe_clusters()
+
+for cluster in clusters['Clusters']:
+    if 'KmsKeyId' not in cluster or not cluster['KmsKeyId']:
+        print(f"Cluster {cluster['ClusterIdentifier']} is not encrypted with a KMS CMK.")
+```
+
+This script will print out the identifiers of all Redshift clusters that are not encrypted with a KMS CMK.
+</Accordion>
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>